var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -140,14 +151,10 @@ def mpi_dependent_options(self):
                 raise RuntimeError('\n'.join(errors))
         else:
             compiler_opts = [
-                '--with-mpi=1',
-                '--with-mpi-dir=%s' % self.spec['mpi'].prefix,
+                '--with-cc=%s' % self.spec['mpi'].mpicc,
+                '--with-cxx=%s' % self.spec['mpi'].mpicxx,
+                '--with-fc=%s' % self.spec['mpi'].mpifc
             ]
-        if sys.platform != "darwin":
-            compiler_opts.extend([
-                '--with-cpp=cpp',
```

Comment:
@eschnett had problems on his HPC and added those lines in https://github.com/LLNL/spack/pull/1807 . Do you have some issues when `--with-cpp` is set?